pyNastran/bdf/cards/elements/bush.py

```diff
@@ -1,465 +1,467 @@
-        return list_fields
-
-    def write_card(self, size=8, is_double=False):
-        card = self.repr_fields()
-        return self.comment + print_card_8(card)
+# pylint: disable=C0103,R0902,R0904,R0914,C0111
+"""
+All bush elements are defined in this file.  This includes:
+
+ * CBUSH
+ * CBUSH1D
+ * CBUSH2D
+
+All bush elements are BushElement and Element objects.
+"""
+from __future__ import (nested_scopes, generators, division, absolute_import,
+                        print_function, unicode_literals)
+
+from pyNastran.utils import integer_types
+from pyNastran.bdf.field_writer_8 import set_blank_if_default
+from pyNastran.bdf.cards.base_card import Element
+from pyNastran.bdf.bdf_interface.assign_type import (
+    integer, integer_or_blank, integer_double_or_blank, double_or_blank,
+    string_or_blank)
+from pyNastran.bdf.field_writer_8 import print_card_8
+
+
+class BushElement(Element):
+    def __init__(self):
+        self.cid = None
+        Element.__init__(self)
+
+    def Cid(self):
+        if self.cid is None:
+            return None
+        elif isinstance(self.cid, integer_types):
+            return self.cid
+        return self.cid_ref.cid
+
+    def Mass(self):
+        return 0.
+
+    def get_edge_ids(self):
+        """
+        Return the edge IDs
+        """
+        return [tuple(sorted(self.node_ids))]
+
+class CBUSH(BushElement):
+    type = 'CBUSH'
+    _field_map = {
+        1: 'eid', 2:'pid', 3:'ga', 4:'gb', 8:'cid', 9:'s', 10:'ocid'
+    }
+
+    def _update_field_helper(self, n, value):
+        if n == 11:
+            self.si[0] = value
+        elif n == 12:
+            self.si[1] = value
+        elif n == 13:
+            self.si[2] = value
+        else:
+            if self.g0 is not None:
+                if n == 5:
+                    self.g0 = value
+                else:
+                    raise KeyError('Field %r=%r is an invalid %s entry.' % (n, value, self.type))
+            else:
+                if n == 5:
+                    self.x[0] = value
+                elif n == 6:
+                    self.x[1] = value
+                elif n == 7:
+                    self.x[2] = value
+                else:
+                    raise KeyError('Field %r=%r is an invalid %s entry.' % (n, value, self.type))
+
+    def __init__(self, eid, pid, ga, gb, x, g0, cid, s, ocid, si, comment=''):
+        BushElement.__init__(self)
+        if comment:
+            self.comment = comment
+        self.eid = eid
+        self.pid = pid
+        self.ga = ga
+        self.gb = gb
+        self.x = x
+        self.g0 = g0
+        self.cid = cid
+        self.s = s
+        self.ocid = ocid
+        self.si = si
+
+    @classmethod
+    def add_card(cls, card, comment=''):
+        eid = integer(card, 1, 'eid')
+        pid = integer_or_blank(card, 2, 'pid', eid)
+        ga = integer(card, 3, 'ga')
+        gb = integer_or_blank(card, 4, 'gb')
+
```

Comment:
@SteveDoyle2 adding comments to let you know where I changed (this line endings are killing me)

Moved the CID block here to use variable cid before the assertion commented below